Index: src/atom.cpp
===================================================================
--- src/atom.cpp	(revision 8387)
+++ src/atom.cpp	(working copy)
@@ -1626,6 +1626,10 @@
   if (strcmp(name,"f") == 0) return (void *) f;
   if (strcmp(name,"mass") == 0) return (void *) mass;
   if (strcmp(name,"rmass") == 0) return (void *) rmass;
+  if (strcmp(name,"omega") == 0) return (void *) omega;
+  if (strcmp(name,"angmom") == 0) return (void *) angmom;
+  if (strcmp(name,"torque") == 0) return (void *) torque;
+  if (strcmp(name,"radius") == 0) return (void *) radius;
 
   return NULL;
 }
